src/main/java/io/fabric8/maven/docker/config/RegistryAuthConfiguration.java

```diff
@@ -38,16 +42,23 @@ public Map toMap() {
             authMap.put("pull", pull);
         }
         if (StringUtils.isNotBlank(username)) {
-            authMap.put(AuthConfigFactory.AUTH_USERNAME, username);
+            authMap.put(AuthConfig.AUTH_USERNAME, username);
         }
         if (StringUtils.isNotBlank(password)) {
-            authMap.put(AuthConfigFactory.AUTH_PASSWORD, password);
+            authMap.put(AuthConfig.AUTH_PASSWORD, password);
         }
+
+        // deprecated "authToken"
         if (StringUtils.isNotBlank(authToken)) {
-            authMap.put(AuthConfigFactory.AUTH_AUTHTOKEN, authToken);
+            authMap.put(AuthConfig.AUTH_AUTH, authToken);
         }
+        // overwrite with "auth"
+        if (StringUtils.isNotBlank(auth)) {
```

Comment:
This will certainly make do, but I suggest to throw a _configuration exception_ if both auth & authToken are set.
This may seem disruptive, yet: if someone specified both, they obviously worked on that part just now and would thus be grateful to know that there's configuration they can remove.